### PR TITLE
git: wrap git-credential-netrc to set PERL5LIB

### PR DESCRIPTION
**Pull request by cyounkins**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/48199

@manveru @volth 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/git/default.nix [hunk 1]
```diff
@@ -178,6 +177,11 @@ stdenv.mkDerivation {
       for i in ${builtins.toString perlLibs}; do
         gitperllib=$gitperllib:$i/lib/perl5/site_perl
       done
+
+      ln -s "$out/share/git/contrib/credential/netrc/git-credential-netrc" $out/bin/
+      wrapProgram $out/bin/git-credential-netrc \
+                  --set PERL5LIB "$gitperllib"
```
Comment (volth) [inline pkgs/applications/version-management/git-and-tools/git/default.nix]:
```
makeWrapper "$out/share/git/contrib/credential/netrc/git-credential-netrc" $out/bin/git-credential-netrc \
  --set PERL5LIB "$gitperllib"
```
could replace both ```ln``` and ```wrapProgram``` (but I have not tested)

Comment (cyounkins) [reply pkgs/applications/version-management/git-and-tools/git/default.nix]: Thanks!

https://nixos.org/nixpkgs/manual/#ssec-stdenv-functions says "wrapProgram is a convenience function you probably want to use most of the time." I'm new to nixpkgs - could you explain why you'd prefer makeWrapper here?

**Comment by c0bw3b** [comment]
Could you also target this at the `staging` branch?

**Comment by cyounkins** [comment]
When basing the diff off the `staging` branch, I'm getting this error when trying to install using the local clone: `trace: version 19.03.git.6f2475f does not appear in name nixos-version`

Full output: https://gist.github.com/cyounkins/4d8c4ced79742c47d01586078f00dbea

Am I doing something stupid? I'm doing pretty much what is in https://nixos.wiki/wiki/Nixpkgs/Create_and_debug_packages

**Comment by cyounkins** [comment]
Oops, sorry for the noise. Now targeting `staging`

**Pull request merged**